feedback/api/views.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import viewsets
+from rest_framework.permissions import(
+	AllowAny,
+)
+from feedback.api.serializers import FeedbackSerializer
+from feedback.models import Feedback
+from rest_framework.filters import (
+	SearchFilter,
+	OrderingFilter,
+	)
+from .permissions import IsOwnerFeedback
+from django_filters import rest_framework as filters
+from rest_framework.response import Response
+from rest_framework import status
+from courses.api.pagination import ResourcesPagination
+
+
+class FeedbackViewSet(viewsets.ModelViewSet):
+	"""
+	handles viewset for feedback serializer.
+	"""
+	serializer_class = FeedbackSerializer
+	pagination_class = ResourcesPagination
+	permission_classes = [AllowAny,IsOwnerFeedback]
+	filter_backends = [filters.DjangoFilterBackend,SearchFilter, OrderingFilter]
+	search_fields = ['id','text']
+	ordering = ('-created_at')
+	queryset = Feedback.objects.all()
+
+	def create(self, request, *args, **kwargs):
+		serializer = self.get_serializer(data=request.data)
+		serializer.is_valid(raise_exception=True)
+		self.perform_create(serializer)
+		headers = self.get_success_headers(serializer.data)
+		return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+	def perform_create(self, serializer):
+		if self.request.user.id == None:
```

Comment:
How does the `request.user` look for an Anonymous user?

Reply:
If there is an Anonymous user then the value for user column would be Null.
i.e there will be no id assigned to it.
So this perform create method first checks if there is a user with no id i.e 
"""
if self.request.user.id == None
"""
if there is no value in the user column then save the serializer with Null as the value for user column else save the user with id. 
If you test the code you will see that an anonymous 
user's object, the value against user column is null

Reply:
No. What comes in the request. print out on the console and share it.

`print request.user.__dict__` for anonymous requests

Reply:
if I print #print(self.request.user.id) when any user is logged in it prints user's id in the console
and in the case of the anonymous user, it prints None.
But if I print #print(self.request.user.__dict__)  when any user is logged in it prints something like
{'_setupfunc': <function AuthenticationMiddleware.process_request.<locals>.<lambda> at 0x000001EA2E879950>, '_wrapped': <User: psp>} 
and in the case of anonymous user it prints out empty dict.

Reply:
I think that's because the request is made by the user who is not stored in the database